src/main/app/controller/PostController.ts

```diff
@@ -37,6 +37,8 @@ export class PostController<T extends AnyObject> {
       await this.saveAndSignOut(req, res, stepData);
     } else if (req.body.saveBeforeSessionTimeout) {
       await this.saveBeforeSessionTimeout(req, res, stepData);
+    } else if (req.session.isApplicant2) {
```

Comment:
i think you can have an if statement before this with this check but just sets eventName to CITIZEN_APPLICANT2_UPDATE so you dont need another if statement here to call the same method as below.